### PR TITLE
Implement recovery compasses

### PR DESCRIPTION
**Pull request by JustTalDevelops**

### CODE REVIEW

File: server/player/player.go [hunk 1]
```diff
@@ -89,6 +89,11 @@ type Player struct {
 	lastXPPickup atomic.Value[time.Time]
 	immunity     atomic.Value[time.Time]
 
+	deathMu        sync.Mutex
+	hasDied        bool
+	deathPos       mgl64.Vec3
+	deathDimension world.Dimension
```
Comment (Sandertv) [inline server/player/player.go:95]: Would just make one of these fields a pointer so you can remove the `deathPos` field.

**Review by Sandertv — COMMENTED** [review]
Looks good. I've got one small nitpick to get rid of one of the fields, but this should be good to go afterwards.

File: server/player/player.go (hunk 1 above)
Comment (JustTalDevelops) [reply server/player/player.go:95]: Did you mean the `hasDied` field?

Comment (Sandertv) [reply server/player/player.go:95]: I did, sorry!

Comment (JustTalDevelops) [reply server/player/player.go]: Removed.

**Review by Sandertv — APPROVED** [review]
LGTM! Thanks for the PR!

**Pull request merged**